src/lib.rs

```diff
@@ -208,9 +208,108 @@
 // Ensure we're `no_std` when compiling for Wasm.
 #![cfg_attr(not(feature = "std"), no_std)]
 
-use srml_support::traits::Currency;
-use srml_support::{decl_event, decl_module, decl_storage, dispatch, StorageValue};
-use system::ensure_signed;
+#[cfg(feature = "std")]
+#[macro_use]
+extern crate serde_derive;
+
+use rstd::prelude::*;
+
+use parity_codec_derive::{Decode, Encode};
+use srml_support::{ decl_event, decl_module, decl_storage, ensure, dispatch, StorageValue, StorageMap};
+
+/// Length constraint for input validation
+
+enum LengthValidationResult {
+    TooShort,
+    TooLong,
+    Success
+}
+
+struct InputValidationLengthConstraint {
+
+    /// Minimum length
+    min : usize,
+
+    /// Maximum length
+    /// While having max would have been more direct, this
+    /// way makes max < min unrepresentable semantically, 
+    /// which is safer.
+    max_as_min_diff: usize,
+}
+
+impl InputValidationLengthConstraint {
+    
+    /// Helper for computing max
+    fn max(&self) -> usize {
+        self.min + self.max_as_min_diff
+    }
+
+    /// Just to give method interface to read min, like max
+    fn min(&self) -> usize {
+        self.min
+    }
+
+    fn validate(&self, length: usize) -> LengthValidationResult {
+
+        if length < self.min {
+            LengthValidationResult::TooShort
+        }
+        else if length > self.max() {
+            LengthValidationResult::TooLong
+        }
+        else {
+            LengthValidationResult::Success
+        }
+    } 
+
+    // TODO: add more vliadtion stuff here in the future?
+}
+
+/// Constants
+const CATEGORY_TITLE: InputValidationLengthConstraint = InputValidationLengthConstraint{
+    min: 3,
+    max_as_min_diff: 30
+};
+
+const CATEGORY_DESCRIPTION: InputValidationLengthConstraint = InputValidationLengthConstraint{
+    min: 10,
+    max_as_min_diff: 100
+};
+
+/// The greatest valid depth of a category.
+/// The depth of a root category is 0.
+const MAX_VALID_CATEGORY_DEPTH: u32 = 3;
```

Comment:
This name sounds more user-friendly to me, IMHO :)
```suggestion
const MAX_NUMBER_OF_SUBCATEGORIES: u32 = 3;
```

Reply:
I think `MAX_NUMBER_OF_SUBCATEGORIES` is ambiguous, it can be interpreted as 

"In one category, you can only have max 3 categories inside it". 

This is why I use the term "depth", because it has a clear meaning in the context of a tree, it refers to multiple levels.

Changing to `MAX_CATEGORY_DEPTH`, I think its quite clear. Also comment explains

```
/// The greatest valid depth of a category.
/// The depth of a root category is 0.
```

Anyway, not sure what you mean by "user", but developers get it, end users don't read these names, they will have some nice explanation in Pioneer or similar :)